mitodl/open-discussions
# Additions to Profile model and DRF API

#### What are the relevant tickets?
Closes #674

#### What's this PR do?
- Adds `bio`, `headline`, and `toc_optin` fields to the `profiles.models.Profile` model
- Adds `profiles.views.ProfileViewSet`
- Adds profile permissions
- Updates `profiles.serializers.ProfileSerializer`

#### How should this be manually tested?
- `/api/v0/users` should return a list of users including their profiles, with each profile including `bio` and `headline`, but you'll need proper authentication first:
  - In a django shell, get a staff-role token:
    ```
    from fixtures.users import staff_jwt_token
    from django.contrib.auth.models import User
    user = User.objects.get(email=<your_user_email>)
    staff_jwt_token(None, user)
    ```
   - Using Postman or curl, send a GET request to `/api/v0/users` with Authentication header `Bearer <token_value>`
    <img width="703" alt="screen shot 2018-05-10 at 11 13 14 am" src="https://user-images.githubusercontent.com/187676/39877170-301d9eba-5443-11e8-8106-8d4739538517.png">

- Go to `/api/v0/profiles/<your_username>/`.  Make changes in the form and submit.  The changes should be saved.
- Go to `/api/v0/profiles/<another_username>/`.  There should not be any form to submit.
- Log in via `/admin/login` as a superuser and go to `/api/v0/profiles/<another_username>/`.  Make changes in the form and submit.  The changes should be saved.
